### PR TITLE
#6594 allow to publish personal gists

### PR DESCRIPTION
**Pull request by MariuszJurowicz**
Wait till https://github.com/twosigma/beakerx/pull/6628 is merged.

### CODE REVIEW

**Comment by scottdraves** [comment]
cool, that works great!
however we need the access token to be persistent, stored in beakerx.json like the JVM options are.

**Comment by MariuszJurowicz** [comment]
@scottdraves 
I'm not sure how the use case for storing the Personal Access Token should look like.
It can be useful if the single instance of beakerx is used by single user, but as we don't introduce user accounts it can be the security breach.

Maybe we can add the `remember` checkbox and store the token in session storage or local storage.

**Comment by MariuszJurowicz** [comment]
UC: if user fills the token input, store it in beakerx.json and fill the input next time with the saved token. If user removes the token from input, clear the beakerx.json token that was saved before. If user fills the input with different token, overwrite the saved token with the new one.

**Comment by jmsdnns** [comment]
You are right to be concerned about the personal access token being something that could be hijacked.

Do we do anything to help the user limit the scope of the personal access token? We only need the "create gists" scope, which would protect everything else in the user's github account.

You can see all the scopes here and see that "create gists" is one of the choices: https://github.com/settings/tokens/new

**Comment by MariuszJurowicz** [comment]
@jmsdnns @scottdraves 
I've added the informatio about scopes to publish modal:

> Enter your Personal Access Token to publish gist to your github account.
> Make sure your Personal Access Token has the _gists_ scope _only_.
> You can read about scopes [here](https://developer.github.com/apps/building-oauth-apps/scopes-for-oauth-apps/)

File: js/notebook/src/extension/gistPublish/gistPublishModal.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  static create(submitCallback, personalAccessToken = '') {
+    const modalContent = GistPublishModal.createModalContent();
+    const personalAccessTokenInput = modalContent.querySelector('input');
+
+    if (personalAccessTokenInput) {
+      personalAccessTokenInput.value = personalAccessToken;
+    }
+
+    dialog.modal({
+      title : 'Publish to a Github Gist',
```
Comment (scottdraves) [inline js/notebook/src/extension/gistPublish/gistPublishModal.ts]: should be capitalized "GitHub"

**Comment by scottdraves** [comment]
screen shot 2018-01-19 at 9 13 01 am
screen shot 2018-01-19 at 9 13 17 am

**Comment by scottdraves** [comment]
actually, i had forgotten to add the "gist" scope to the token.
so the error is not that informative but this is in fact the error github reports, so i don't see a good way to fix this.  the good news is, the error is reported nicely :)

**Pull request merged**